src/constants/contracts.ts

```diff
@@ -82,3 +86,17 @@ export const FIATDAO_WSOHM_CONTRACT = new Contract({
   name: "FiatDAO Contract",
   addresses: FIATDAO_WSOHM_ADDRESSES,
 });
+
+export const BALANCER_VAULT = new Contract({
+  factory: BalancerVault__factory,
+  name: "Balancer Vault Contract",
+  addresses: BALANCER_VAULT_ADDRESSSES,
+});
+
+export const CURVE_FACTORY = new Contract({
+  factory: CurveFactory__factory,
+  name: "Curve Factory Contract",
+  addresses: {
+    [NetworkId.MAINNET]: "0xf18056bbd320e96a48e3fbf8bc061322531aac99",
+  },
```

Comment:
same feedback as above. 

Curve factory retrieved from contract call: 

```
  async _curveTokenAddresses(contract: CurveETHSwapPool) {
    const factoryAddress = await contract.factory();
    const factoryContract = CurveFactory__factory.connect(factoryAddress, this.provider);
    const tokens = await factoryContract.get_coins(contract.address);
    return tokens;
  }
```

Reply:
Correct me if i'm wrong here, but I noticed that all Curve factory pools are deployed from this single factory contract, meaning the `factoryAddress` returned here is the same for every pool. That's why I added it as a static contract, since it won't ever change and allows us to avoid an unnecessary request.

Reply:
that seems true for now, but it would be nice if this was fully automated & we didn't have to check for a new factory address in the future.

Reply:
Done